light_curves/code_src/ztf_functions.py

```diff
@@ -1,325 +1,106 @@
-        result = tap_service.run_async(query, uploads={"sample": upload_table[i: i + chunksize]})
-        locations.append(result.to_table().to_pandas())
-
-    # locations may contain more than one ZTF object id per band (e.g., yang sample sample_table[11])
-    # Sánchez-Sáez et al., 2021 (2021AJ....162..206S)
-    # return all the data -- transform_lightcurves will choose which to keep
-    return pd.concat(locations, ignore_index=True)
-
-
-def load_lightcurves(locations_df, nworkers=6, chunksize=100):
-    """Loop over the catalog's parquet files (stored in an AWS S3 bucket) and load light curves.
-
-    Parameters
-    ----------
-    locations_df : pd.DataFrame
-        Dataframe with ZTF field, CCD, quadrant and other information that identifies each `coord` object
-        and which parquet files it is in.
-    nworkers : int or None
-        Number of workers in the multiprocessing pool. Use None to turn off multiprocessing. This must be None
-        if this function is being called from within a child process (no nested multiprocessing).
-    chunksize : int
-        Number of files sent to the workers.
-
-    Returns
-    -------
-    ztf_df : pd.DataFrame
-        Dataframe of light curves. Expect one row per oid in locations_df. Each row
-        stores a full light curve. Elements in the columns "mag", "hmjd", etc. are arrays.
-    """
-    # We need to return an empty dataframe if no matches are found. If the TAP query returned matches
-    # but none of them are found in the parquet files, this function will naturally return an empty dataframe.
-    # But if the TAP query found no matches, pd.concat (below) will throw a ValueError. Return now to avoid this.
-    if len(locations_df.index) == 0:
-        return pd.DataFrame()
-
-    # If CATALOG_FILES hasn't been loaded yet, do it now. We'll use the checksums file to get the list.
-    global CATALOG_FILES
-    if CATALOG_FILES is None:
-        CATALOG_FILES = (
-            pd.read_table(f"s3://{CATALOG_ROOT}checksum.md5", sep="\s+",
-                          names=["md5", "path"], usecols=["path"])
-            .squeeze()  # there's only 1 column. squeeze it into a Series
-            .str.removeprefix("./")
-            .to_list()
+    MultiIndexDFObject
+     """
+    # 1) Start Dask client & read full ZTF light-curve catalog
+    client = Client(threads_per_worker=2, memory_limit=None)
+    suffix = 'ztf_lc_dr23'
+    ztf_lc = lsdb.read_hats(
+        UPath('s3://irsa-fornax-testdata/ZTF/dr23/lc/'),
+        columns=[
+            "objectid", "objra", "objdec",
+            "hmjd", "filterid", "mag", "magerr", "catflags"
+        ]
+    )
+
+    # 2) Convert Astropy table → pandas → LSDB catalog
+    sample_df = pd.DataFrame({
+        'objectid': sample_table['objectid'],
+        'ra_deg':   sample_table['coord'].ra.deg,
+        'dec_deg':  sample_table['coord'].dec.deg,
+        'label':    sample_table['label'],
+    })
+    sample_lsdb = lsdb.from_dataframe(
```

Comment:
would be a nice upstream feature to be able to ingest directly from an astropy table. Has that been raised?

Reply:
That would be nice. I would be surprised if truly no one has requested it yet but I haven't heard anything about it. A quick look through their issues didn't turn up anything obviously related so probably worth opening an issue for.

Reply:
looks like @bsipocz did this.  Issue 780 over at lsdb

Reply:
Yes, sorry, I forgot to xref it here: https://github.com/astronomy-commons/lsdb/issues/780